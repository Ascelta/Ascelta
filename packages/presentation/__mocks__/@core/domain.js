export class SuiteUser {
  constructor(vUserDetail) {
    this.vUserDetail = vUserDetail;
  }
}

export const AuthStatus = {
  LOADING: 'loading',
  SIGNED_IN: 'signedIn',
  SIGNED_OUT: 'signedOut',
};