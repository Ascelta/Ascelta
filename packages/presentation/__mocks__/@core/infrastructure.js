export const authRepository = {
  onAuthStateChange: jest.fn(),
};