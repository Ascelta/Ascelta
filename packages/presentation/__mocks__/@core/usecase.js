export class CheckScreenNameExistenceUseCase {}
export class FindSuiteUserUseCase {}
export class GetCurrentUserIdUseCase {}
export class SignInUseCase {}
export class SignOutUseCase {}
export class UpdateScreenNameUseCase {}
export class UpdateUserProfileUseCase {}