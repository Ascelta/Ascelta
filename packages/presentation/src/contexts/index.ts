export * from './AuthContext';
export * from './UseCaseContext';
