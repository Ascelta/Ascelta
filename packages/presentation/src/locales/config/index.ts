import { initReactI18next } from 'react-i18next';
import { getLocales } from 'react-native-localize';
import i18n from 'i18next';
import en from './../en.json';
import ja from './../ja.json';

export const supportedLngs = {
  en: 'English',
  ja: '日本語',
};

i18n.use(initReactI18next).init({
  supportedLngs: Object.keys(supportedLngs),
  resources: {
    en: { translation: en },
    ja: { translation: ja },
  },
  lng: getLocales()[0].languageCode,
  fallbackLng: ['en', 'ja'],
  interpolation: {
    escapeValue: false,
  },
});

export default i18n;
