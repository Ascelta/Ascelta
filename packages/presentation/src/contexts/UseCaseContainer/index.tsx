import React, { ReactNode, createContext, useContext } from 'react';
import { SignInUseCase } from '@core/usecase';
import { useCaseContainer } from '../../composition';

export interface UseCaseContainer {
  readonly signInUseCase: SignInUseCase;
}

const UseCaseContext = createContext<UseCaseContainer>(useCaseContainer);

type ProviderProps = {
  container?: Partial<UseCaseContainer>;
  children: ReactNode;
};
export const UseCaseProvider: React.FC<ProviderProps> = ({ container = {}, children }) => {
  const value = { ...useCaseContainer, ...container };
  return <UseCaseContext.Provider value={value}>{children}</UseCaseContext.Provider>;
};

// カスタムフック
export function useUseCases(): UseCaseContainer {
  return useContext(UseCaseContext);
}
