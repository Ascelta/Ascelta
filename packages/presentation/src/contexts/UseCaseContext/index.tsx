import React, { ReactNode, createContext, useContext } from 'react';
import { FindSuiteUserUseCase, GetCurrentUserIdUseCase, SignInUseCase, SignOutUseCase } from '@core/usecase';
import { useCaseContainer } from '../../composition';

export interface UseCaseContainer {
  readonly findSuiteUserUseCase: FindSuiteUserUseCase;
  readonly getCurrentUserIdUseCase: GetCurrentUserIdUseCase;
  readonly signInUseCase: SignInUseCase;
  readonly signOutUseCase: SignOutUseCase;
}

const UseCaseContext = createContext<UseCaseContainer>(useCaseContainer);

type ProviderProps = {
  container?: Partial<UseCaseContainer>;
  children: ReactNode;
};
export const UseCaseProvider: React.FC<ProviderProps> = ({ container = {}, children }) => {
  const value = { ...useCaseContainer, ...container };
  return <UseCaseContext.Provider value={value}>{children}</UseCaseContext.Provider>;
};

export const useUseCases = () => useContext(UseCaseContext);
