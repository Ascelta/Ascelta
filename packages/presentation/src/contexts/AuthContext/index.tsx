import React, { PropsWithChildren, createContext, useContext, useEffect, useState } from 'react';
import { authRepository } from '@core/infrastructure';
import { AuthProviderType } from '@core/domain';
import { useUseCases } from '../UseCaseContext';

export interface Props {
  userId: string | undefined;
  signIn: (provider: AuthProviderType) => Promise<void>;
  signOut: () => Promise<void>;
}

const AuthContext = createContext<Props | undefined>(undefined);

export const AuthProvider: React.FC<PropsWithChildren> = ({ children }) => {
  const { getCurrentUserIdUseCase, signInUseCase, signOutUseCase } = useUseCases();
  const [userId, setUserId] = useState<string | undefined>(undefined);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    (async () => {
      const userId = await getCurrentUserIdUseCase.execute();
      setUserId(userId);
      setLoading(false);
    })();
    return authRepository.onAuthStateChange((status, userId) => {
      setUserId(userId);
      setLoading(false);
    });
  }, []);

  const value: Props = {
    userId: userId,
    signIn: async type => signInUseCase.execute(type),
    signOut: async () => signOutUseCase.execute(),
  };

  return <AuthContext.Provider value={value}>{!loading && children}</AuthContext.Provider>;
};

export const useAuth = () => {
  const context = useContext(AuthContext);
  if (!context) {
    throw new Error('useAuth must be used within an AuthProvider');
  }
  return context;
};
