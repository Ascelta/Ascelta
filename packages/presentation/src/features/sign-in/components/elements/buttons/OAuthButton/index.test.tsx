import '@testing-library/jest-dom';
import { act, fireEvent, render, screen } from '@testing-library/react';
import { TamaguiProvider } from 'tamagui';
import { SignInUseCase } from '@core/usecase';
import { AuthRepository } from '@core/domain';
import config from '../../../../../../../tamagui.config.ts';
import { UseCaseProvider } from '../../../../../../contexts/UseCaseContainer';

jest.mock('solito/router', () => {
  // モックファクトリ内で直接作る
  const replaceMock = jest.fn();
  return {
    __esModule: true,
    useRouter: jest.fn().mockImplementation(() => ({
      replace: replaceMock,
    })),
    // 必要ならテスト内で import して使えるようエクスポート
    replaceMock: replaceMock,
  };
});

jest.mock('@core/usecase', () => {
  const executeMock = jest.fn<Promise<void>, [string]>();
  return {
    __esModule: true,
    SignInUseCase: jest.fn().mockImplementation(() => ({
      execute: executeMock,
    })),
    executeMock: executeMock,
  };
});

const executeMock = require('@core/usecase').executeMock;

jest.mock('../../../../../../composition', () => ({
  useCaseContainer: {},
}));

const replaceMock = require('solito/router').replaceMock;

const OAuthButton = require('./index.tsx').OAuthButton;

describe('<OAuthButton />', () => {
  const setup = () => {
    return render(
      <UseCaseProvider
        container={{
          signInUseCase: new SignInUseCase({} as AuthRepository),
        }}
      >
        <TamaguiProvider config={config} defaultTheme='light'>
          <OAuthButton type='google' icon={<div />} backgroundColor='#FFF' borderColor='#FFF' text='Sign in with Google' textColor='#000' />
        </TamaguiProvider>
      </UseCaseProvider>,
    );
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('スナップショットが一致すること', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });

  it('ボタンをタップした場合は、SignInUseCase#execute が呼ばれていること', async () => {
    setup();
    const button = screen.getByText('Sign in with Google');
    expect(button).toBeInTheDocument();
    expect(executeMock).not.toHaveBeenCalled();
    expect(replaceMock).not.toHaveBeenCalled();
    await act(async () => {
      fireEvent.click(button);
    });
    expect(executeMock).toHaveBeenCalledTimes(1);
    expect(executeMock).toHaveBeenCalledWith('google');
    expect(replaceMock).toHaveBeenCalledTimes(1);
    expect(replaceMock).toHaveBeenCalledWith('/home');
  });
});
