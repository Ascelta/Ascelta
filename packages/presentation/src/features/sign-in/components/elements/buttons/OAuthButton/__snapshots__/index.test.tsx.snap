// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OAuthButton /> スナップショットが一致すること 1`] = `
<DocumentFragment>
  <span
    class=""
    style="display: contents;"
  >
    <button
      class="is_Button _outlineStyle-0focus-visible-solid _outlineWidth-0focus-visible-2px _bg-0hover-FFF35 _btc-0hover-FFF35 _brc-0hover-FFF35 _borderBottomColor-0hover-FFF35 _borderLeftColor-0hover-FFF35 _bg-0active-FFF35 _btc-0active-FFF35 _brc-0active-FFF35 _borderBottomColor-0active-FFF35 _borderLeftColor-0active-FFF35 _bg-0focus-FFF35 _btc-0focus-FFF35 _brc-0focus-FFF35 _borderBottomColor-0focus-FFF35 _borderLeftColor-0focus-FFF35 _dsp-flex _items-center _flexBasis-auto _bxs-border-box _pos-relative _minH-0px _minW-0px _shrink-0 _flexDirection-row _pr-t-space-tru101 _pl-t-space-tru101 _height-t-size-true _btlr-t-radius-6 _btrr-t-radius-6 _borderBottomRightRadius-t-radius-6 _borderBottomLeftRadius-t-radius-6 _justify-center _fwr-nowrap _cur-pointer _bg-FFF35 _btw-1px _brw-1px _borderBottomWidth-1px _borderLeftWidth-1px _btc-FFF35 _brc-FFF35 _borderBottomColor-FFF35 _borderLeftColor-FFF35 _borderBottomStyle-solid _borderTopStyle-solid _borderLeftStyle-solid _borderRightStyle-solid"
      data-disable-theme="true"
      role="button"
      tabindex="0"
    >
      <div
        class="_dsp-flex _flexBasis-auto _bxs-border-box _pos-relative _minH-0px _minW-0px _shrink-1 _flexDirection-row _grow-1 _width-10037 _items-center _justify-center"
      >
        <div
          class="_dsp-flex _flexDirection-column _flexBasis-auto _bxs-border-box _pos-relative _minH-0px _minW-0px _shrink-0 _width-24 _height-24 _justify-center _items-center"
        >
          <div />
        </div>
        <span
          class="is_Spacer _dsp-flex _items-stretch _flexDirection-column _flexBasis-auto _bxs-border-box _pos-relative _minH-t-space-6 _minW-t-space-6 _shrink-0 _pe-none _width-t-space-6 _height-t-space-6"
        />
        <div
          class="_dsp-flex _items-stretch _flexDirection-column _flexBasis-auto _bxs-border-box _pos-relative _minH-0px _minW-0px _shrink-0"
        >
          <p
            class="is_Paragraph font_body _dsp-inline _bxs-border-box _ww-break-word _mt-0px _mr-0px _mb-0px _ml-0px _ff-f-family _fw-bold _ls-f-letterSpa1360334201 _fs-f-size-5 _lh-f-lineHeigh112924 _col-00035 _select-auto _ws-normal"
          >
            Sign in with Google
          </p>
        </div>
        <span
          class="is_Spacer _dsp-flex _items-stretch _flexDirection-column _flexBasis-auto _bxs-border-box _pos-relative _minH-24 _minW-24 _shrink-0 _pe-none _width-24 _height-24"
        />
      </div>
    </button>
    <div
      style="display: contents;"
    />
  </span>
</DocumentFragment>
`;
